src/annotator/test/guest-test.js

```diff
@@ -420,15 +420,18 @@ describe('Guest', () => {
     });
 
     describe('on "deleteAnnotation" event', () => {
-      it('detaches annotation', () => {
```

Comment:
This test replaces a test for the main scenario of using `deleteAnnotation` with one that handles this edge case. I think it would make sense to have both.